src/util/pm_buffer.c

```diff
@@ -151,6 +151,16 @@ pm_buffer_append_varuint(pm_buffer_t *buffer, uint32_t value) {
     }
 }
 
+/**
+ * Append a 32-bit signed integer to the buffer as a variable-length integer.
+ */
+void
+pm_buffer_append_varsint(pm_buffer_t *buffer, int32_t value) {
+    int64_t large_value = (int64_t)value;
+    uint32_t unsigned_int = (uint32_t)((2 * large_value) ^ (large_value >> (sizeof(int32_t) * 8 - 1)));
```

Comment:
Is the conversion to `int64_t` needed?
The >> 31 shouldn't need it and is probably clearer without since it's for getting the sign bit.
The << 1 is probably fine too (https://protobuf.dev/programming-guides/encoding/#signed-ints).
So I'm thinking:
```c
uint32_t unsigned_int = ((uint32_t)(value) << 1) ^ (uint32_t)(value >> 31));
```
I found https://github.com/protocolbuffers/protobuf/blob/842f56b57a9e825d6cdd6c00947f6c19c18fe466/src/google/protobuf/io/coded_stream.h#L886-L887 so it seems correct.

Reply:
TIL about `>>>`.

Done both.